instrumentation/micrometer/micrometer-1.5/library/src/main/java/io/opentelemetry/instrumentation/micrometer/v1_5/OpenTelemetryMeterRegistryBuilder.java

```diff
@@ -38,12 +49,27 @@ public OpenTelemetryMeterRegistryBuilder setBaseTimeUnit(TimeUnit baseTimeUnit)
     return this;
   }
 
+  /**
+   * Enables the "Prometheus mode" - this will simulate the behavior of Micrometer's {@code
+   * PrometheusMeterRegistry}. The instruments will be renamed to match Micrometer instrument
+   * naming, and the base time unit will be set to seconds.
+   *
+   * <p>Set this to {@code true} if you are using the Prometheus metrics exporter.
+   */
+  public OpenTelemetryMeterRegistryBuilder setPrometheusMode(boolean prometheusMode) {
+    this.prometheusMode = prometheusMode;
+    return setBaseTimeUnit(TimeUnit.SECONDS);
```

Comment:
instead of setting baseTimeUnit here, which makes the final result dependent on the ordering of calling the setters, what do you think of initializing baseTimeUnit to `null` and doing the calculation in `build()`

Reply:
Good point, will do 👍